modules/webgl/src/classes/program.js

```diff
@@ -185,36 +188,64 @@ export default class Program extends Resource {
     return true;
   }
 
-  setUniforms(uniforms = {}, _onChangeCallback = () => {}) {
-    // Simple change detection - if all uniforms are unchanged, do nothing
-    let somethingChanged = false;
-    const changedUniforms = {};
-    for (const key in uniforms) {
-      if (!areUniformsEqual(this.uniforms[key], uniforms[key])) {
```

Comment:
So this check is turning out to be more expensive than actually setting the uniforms?

Reply:
I've moved the check to inside the setter: https://github.com/uber/luma.gl/pull/1283/files#diff-a257102d168076156bae989f2d59cdb1R296

It is much faster this way.